File: API/ContainerNinja.Contracts/DTO/CalledIngredientDetailsDTO.cs
using ContainerNinja.Contracts.Data.Entities;

namespace ContainerNinja.Contracts.DTO
{
    public class CalledIngredientDetailsDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public KitchenProductDTO KitchenProduct { get; set; }
        public float? Amount { get; set; }
        public int KitchenUnitType { get; set; }
        public int KitchenProductId { get; set; }
        //public KitchenUnitType KitchenUnitType { get; set; }

        public IList<KitchenProductDTO> KitchenProductSearchItems { get; set; } = new List<KitchenProductDTO>();
    }
}

File: Client/src/app/models/CalledIngredientDetailsDTO.ts
import { KitchenProductDTO } from "./KitchenProductDTO";

export class CalledIngredientDetailsDTO {
  id?: number = undefined;
  name?: string = undefined;
  kitchenProduct?: KitchenProductDTO = undefined;
  amount?: number | undefined = undefined;
  kitchenUnitType?: number = undefined;
  kitchenProductId?: number = undefined;
  kitchenProductSearchItems?: KitchenProductDTO[] = undefined;
}
